src/stripeLogsView.ts

```diff
@@ -1,15 +1,203 @@
+      switch (this.viewState) {
+        case ViewState.Idle:
+          return {
+            label: 'Start streaming API logs',
+            command: 'startLogsStreaming',
+            iconId: 'play-circle',
+          };
+        case ViewState.Loading:
+          return {
+            label: 'Starting streaming API logs...',
+            command: 'stopLogsStreaming',
+            iconId: 'loading',
+          };
+        case ViewState.Streaming:
+          return {
+            label: 'Stop streaming API logs',
+            command: 'stopLogsStreaming',
+            iconId: 'stop-circle',
+          };
+      }
+    })();
+
+    const streamingControlItem = this.createItemWithCommand(streamingControlItemArgs);
+
+    const treeItems = [streamingControlItem];
+
+    if (this.logTreeItems.length > 0) {
+      const logsStreamRootItem = new StripeTreeItem('Recent logs');
+      logsStreamRootItem.children = this.logTreeItems;
+      logsStreamRootItem.expand();
+      treeItems.push(logsStreamRootItem);
+    }
+
+    return Promise.resolve(treeItems);
+  }
+
+  private createItemWithCommand({
+    label,
+    command,
+    iconId,
+  }: {
+    label: string;
+    command?: string;
+    iconId?: string;
+  }) {
+    const item = new StripeTreeItem(label, command);
+    if (iconId) {
+      item.iconPath = new ThemeIcon(iconId);
+    }
+    return item;
+  }
+
+  private async setupStreams() {
+    const stripeLogsTailProcess = await this.stripeClient.getOrCreateCLIProcess(
+      CLICommand.LogsTail,
+      ['--format', 'JSON'],
+    );
+    if (!stripeLogsTailProcess) {
+      throw new Error('Failed to start `stripe logs tail` process');
+    }
+
+    stripeLogsTailProcess.on('close', this.cleanupStreams);
+    stripeLogsTailProcess.on('error', this.cleanupStreams);
+
+    // The CLI lets you know that streaming is ready via stderr
+    if (!this.logsStderrStream) {
+      await new Promise<void>((resolve) => {
+        this.logsStderrStream = new stream.Writable({
+          write: (chunk, encoding, callback) => {
+            if (encoding === 'utf8') {
+              if (chunk.includes('Ready!')) {
+                resolve();
+              }
+            }
+            callback();
+          },
+          decodeStrings: false,
+        });
+        stripeLogsTailProcess.stderr
+          .setEncoding('utf8')
+          .pipe(new LineStream())
+          .pipe(this.logsStderrStream);
+      });
+    }
+
+    if (!this.logsStdoutStream) {
+      this.logsStdoutStream = new stream.Writable({
+        write: (chunk, encoding, callback) => {
+          if (encoding === 'utf8') {
+            try {
+              const object = JSON.parse(chunk);
+              if (isLogObject(object)) {
+                const label = `[${object.status}] ${object.method} ${object.url} [${object.request_id}]`;
+                const logTreeItem = new StripeTreeItem(label);
+                this.insertLog(logTreeItem);
+              }
+            } catch {}
```

Comment:
feels like we should be reporting this somewhere, not sure if we need to inform the customer because it's probably not actionable, unless they're using an old/broken cli/extension?

Reply:
Reporting when the output can't be parsed would be great, but sometimes there are stdout messages that we expect to see but can't be JSON.parsed. For example, "Checking for new versions". Is it worth reporting everything that's caught here, or should we do something else?